ote_sdk/ote_sdk/utils/argument_checks.py

```diff
@@ -0,0 +1,206 @@
+"""
+Utils for checking functions and methods arguments
+"""
+
+from os.path import exists
+
+from omegaconf import DictConfig
+from yaml import safe_load
+
+
+def check_parameter_type(parameter, parameter_name, expected_type):
+    """Function raises ValueError exception if parameter has unexpected type"""
+    if not isinstance(parameter, expected_type):
+        parameter_type = type(parameter)
+        raise ValueError(
+            f"Unexpected type of '{parameter_name}' parameter, expected: {expected_type}, actual: {parameter_type}"
+        )
+
+
+def check_required_parameters_type(parameter_name_expected_type: list):
+    """
+    Function raises ValueError exception if required parameters have unexpected type
+    :param parameter_name_expected_type: list with tuples that contain parameter, name for exception message and
+    expected type
+    """
+    for parameter, name, expected_type in parameter_name_expected_type:
+        check_parameter_type(
+            parameter=parameter, parameter_name=name, expected_type=expected_type
+        )
+
+
+def check_optional_parameters_type(parameter_name_expected_type: list):
+    """
+    Function checks if optional parameters exist and raises ValueError exception if one of them has unexpected type
+    :param parameter_name_expected_type: list with tuples that contain optional parameter, name for exception message
+    and expected type
+    """
+    for parameter, name, expected_type in parameter_name_expected_type:
+        if parameter:
```

Comment:
Please, do not use `if parameter` here, since `parameter` may be a valid value that is converted to `False` boolean -- e.g. `0`.
Since `Optional[smthng]` in typing is an alias for `Union[smthng, None]`, it is better to use 
`if parameter is not None` 
instead of `if parameter`.
```suggestion
        if parameter is not None:
```

Reply:
Applied suggestion, thanks